mlrun/utils/vault.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2020 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+import os
+from os.path import expanduser
+import json
+from .helpers import logger
+from ..config import config as mlconf
+from ..k8s_utils import get_k8s_helper
+
+vault_default_prefix = "v1/secret/data"
+vault_url_env_var = "MLRUN_VAULT_URL"
+vault_token_env_var = "MLRUN_VAULT_TOKEN"
+vault_role_env_var = "MLRUN_VAULT_ROLE"
+
+
+class VaultStore:
+    def __init__(self, token=None):
+        self._token = token
+        self.url = mlconf.secret_stores.vault.url or os.environ.get(vault_url_env_var)
```

Comment:
Removed most - I'm keeping the MLRUN_VAULT_TOKEN env variable (also commented in the code on this) since it's mostly for testing/debug purposes and I don't want to formalise it into the config.

Reply:
I still suggest moving it to config, it's just more comfortable when all the configuration is there, and you can easily see what are your options to "set/override" stuff
you can comment near the line in the config, that this is only for debug purposes

Reply:
fine. I'll move it to config.